eras/conway/impl/src/Cardano/Ledger/Conway/Governance.hs

```diff
@@ -204,53 +206,109 @@ toAnchorPairs vote@(Anchor _ _) =
       , "dataHash" .= anchorDataHash
       ]
 
-data Vote era = Vote
-  { voteGovActionId :: !(GovernanceActionId (EraCrypto era))
-  , voteRole :: !VoterRole
-  , voteRoleKeyHash :: !(KeyHash 'Voting (EraCrypto era))
-  , voteAnchor :: !(StrictMaybe (Anchor (EraCrypto era)))
-  , voteDecision :: !VoteDecision
+data VotingProcedure era = VotingProcedure
+  { vProcGovActionId :: !(GovernanceActionId (EraCrypto era))
+  , vProcRole :: !VoterRole
+  , vProcRoleKeyHash :: !(Credential 'Voting (EraCrypto era))
+  , vProcVote :: !Vote
+  , vProcAnchor :: !(StrictMaybe (Anchor (EraCrypto era)))
   }
   deriving (Generic, Eq, Show)
 
-instance NoThunks (Vote era)
+instance NoThunks (VotingProcedure era)
 
-instance Crypto (EraCrypto era) => NFData (Vote era)
+instance Crypto (EraCrypto era) => NFData (VotingProcedure era)
 
-instance Era era => DecCBOR (Vote era) where
+instance Era era => DecCBOR (VotingProcedure era) where
   decCBOR =
     decode $
-      RecD Vote
+      RecD VotingProcedure
         <! From
         <! From
         <! From
-        <! D (decodeNullStrictMaybe decCBOR)
         <! From
+        <! D (decodeNullStrictMaybe decCBOR)
 
-instance Era era => EncCBOR (Vote era) where
-  encCBOR Vote {..} =
+instance Era era => EncCBOR (VotingProcedure era) where
+  encCBOR VotingProcedure {..} =
     encode $
-      Rec (Vote @era)
-        !> To voteGovActionId
-        !> To voteRole
-        !> To voteRoleKeyHash
-        !> E (encodeNullStrictMaybe encCBOR) voteAnchor
-        !> To voteDecision
-
-instance EraPParams era => ToJSON (Vote era) where
-  toJSON = object . toVotePairs
-  toEncoding = pairs . mconcat . toVotePairs
-
-toVotePairs :: (KeyValue a, EraPParams era) => Vote era -> [a]
-toVotePairs vote@(Vote _ _ _ _ _) =
-  let Vote {..} = vote
-   in [ "govActionId" .= voteGovActionId
-      , "role" .= voteRole
-      , "roleKeyHash" .= voteRoleKeyHash
-      , "anchor" .= voteAnchor
-      , "decision" .= voteDecision
+      Rec (VotingProcedure @era)
+        !> To vProcGovActionId
+        !> To vProcRole
+        !> To vProcRoleKeyHash
+        !> To vProcVote
+        !> E (encodeNullStrictMaybe encCBOR) vProcAnchor
+
+instance EraPParams era => ToJSON (VotingProcedure era) where
+  toJSON = object . toVotingProcedurePairs
+  toEncoding = pairs . mconcat . toVotingProcedurePairs
+
+toVotingProcedurePairs :: (KeyValue a, EraPParams era) => VotingProcedure era -> [a]
+toVotingProcedurePairs vProc@(VotingProcedure _ _ _ _ _) =
+  let VotingProcedure {..} = vProc
+   in [ "govActionId" .= vProcGovActionId
+      , "role" .= vProcRole
+      , "roleKeyHash" .= vProcRoleKeyHash
+      , "anchor" .= vProcAnchor
+      , "decision" .= vProcVote
       ]
 
+data ProposalProcedure era = ProposalProcedure
+  { pProcCoin :: !Coin
+  , pProcKeyHash :: !(KeyHash 'Staking (EraCrypto era))
+  , pProcGovernanceAction :: !(GovernanceAction era)
+  , pProcAnchor :: !(StrictMaybe (Anchor (EraCrypto era)))
```

Comment:
Maybe this would be better:
```suggestion
  { pProcDeposit :: !Coin
  , pProcReturnAddr :: !(KeyHash 'Staking (EraCrypto era))
  , pProcGovernanceAction :: !(GovernanceAction era)
  , pProcAnchor :: !(StrictMaybe (Anchor (EraCrypto era)))
```

Reply:
Renamed. Thank you!